File: src/链表/203.移除链表元素/203.移除元素.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    const dummy_head = new ListNode(0, head);
    let cur = dummy_head;
    while(cur.next) {
        if(cur.next.val == val){
            cur.next = cur.next.next;
        } else {
            cur = cur.next;
        }
    }
    return dummy_head.next
};

File: src/链表/203.移除链表元素/203.移除元素.py
# Definition for singly-linked list.

# 定义链表
# class ListNode:
#     def __init__(self, val, next=None):
#         self.val = val
#         self.nect = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy_head = ListNode(next=head)
        cur = dummy_head
        while cur.next != None:
            if cur.next.val == val:
                cur.next = cur.next.next
            else:
                cur = cur.next
        return dummy_head.next





